authentication/views.py

```diff
@@ -1,9 +1,51 @@
-from django.shortcuts import render
-from django.http import HttpResponse
-from django.core import serializers
+import json
+from django.http import JsonResponse
 from authentication.models import *
+from django.contrib.auth import authenticate, login, logout
+from django.views import View
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+import re
 
-# Create your views here.
-def returnTodoJson(request):
-    data = Todo.objects.all()
-    return HttpResponse(serializers.serialize("json", data), content_type="application/json")
+@method_decorator(csrf_exempt, name='dispatch')
+class RegisterView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+        username = data['username']
+        password = data['password']
+        email = data['email']
+        if AppUser.objects.filter(username = username).exists() or AppUser.objects.filter(email = email).exists():
+            return JsonResponse({'msg': 'Username and/or email already taken!'}, status=400)
+        validate_msg = Validators.validateInput(username, email, password)
+        if validate_msg != 'valid':
+            return JsonResponse({'msg': validate_msg}, status=400)
+        new_user = AppUser.objects.create_user(email=email,username=username,password=password)
+        new_user.save()
+        return JsonResponse({'msg': 'Success! User registered'}, status=200)
+
+@method_decorator(csrf_exempt, name='dispatch')
+class LoginView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+        username = data['username']
+        password = data['password']
+        user = authenticate(request, username=username,password=password)
+        if user is not None:
+            login(request, user)
+            return JsonResponse({'msg': 'Success! User logged in'}, status=200)
+        else:
+            return JsonResponse({'msg': 'Wrong username/password!'}, status=400)
```

Comment:
what about using jwt and setting the token expiration within 8 hours? 
ref: https://blog.devgenius.io/jwt-authentication-in-django-rest-framework-with-simple-jwt-a-comprehensive-guide-f2ba860f1365